### ISSUE TITLE
UCUM code for RPM etc?

### ISSUE THREAD

**Issue by dr-shorthair** [ucum-org/ucum#198]
Over in QUDT-land, @jmkeil has proposed to [start using the UCUM code `circ` for 'revolution'](https://github.com/qudt/qudt-public-repo/pull/560) in units like RPM (revs per minute = `circ.min-1`), RPH (`circ.h-1`) etc. 

Was this an intended or anticipated use for `circ`?

**Comment by dr-shorthair** [comment]
Hmm. I get the impression that the UCUM issue-tracker is not being actively monitored by the Advisory Board members - @timbrisc I would like to 'mention' a couple of them in this thread, but can't see their GitHub IDs. 

(My permission level is not high enough to be able to see the teams and people in this repo.)

**Comment by timbrisc** [comment]
Yeah, we may need to call them out. I didn't get notice of your earlier message. @ucum-org/advisors _should_ ping them.

**Comment by dr-shorthair** [comment]
@timbrisc Are you 'watching' the issues list?

**Comment by dr-shorthair** [comment]
So I'll repeat the question, tagging the team this time: 

Over in QUDT-land, @jmkeil has proposed to https://github.com/qudt/qudt-public-repo/pull/560 in units like RPM (revs per minute = circ.min-1), RPH (circ.h-1) etc.

@ucum-org/advisors was this an intended or anticipated use for circ?

**Comment by dr-shorthair** [comment]
Note that `circ` is defined in [UCUM Table 21 – Miscellaneous units](https://ucum.org/ucum#para-47). 

I think it is ultimately related to the question of whether angles are a unit (radians, degrees, revolutions) or a dimensionless ratio (which is likely what the metrologists would argue). Note that I was recently in a CIPM meeting where the issue of angular measurements was raised, and it was agreed that degrees/grads/radians are useful since they indicate a scaling factor on a dimensionless unit. In this context 'revolution' or 'circ' would likely be considered to be useful units.

**Comment by gschadow** [comment]
> Over in QUDT-land, @jmkeil has proposed to [start using the UCUM code `circ` for 'revolution'](https://github.com/qudt/qudt-public-repo/pull/560) in units like RPM (revs per minute = `circ.min-1`), RPH (`circ.h-1`) etc.
> 
> Was this an intended or anticipated use for `circ`?

First of all, who is "Dr. Shorthair"? Is that Simon?

Secondly, those "xPy" unit abbreviations like RPM or my favorite "GPF" (gallons per flush) are not taken on or only very, very reluctantly. We have PPM, yes, but RPM is just /min (per minute) events per minute.

If you defined RPM as circ/min (and I suggest you use the solidus for such simple division, not the -1 exponent, it isn't any less correct for such things), so if you defined RPM as circ/min you have an unexpected kind of quantity, i.e., angular velocity instead of frequency. Most people who need "RPM" probably have no fluent idea what angular velocity even is. 

So, no.

The reason "circ" is there is that it is really the natural base unit of any angle. Gon divides it into 400 units, degrees into 360, and the radians who like to use the arch length of the unit circle instead will divide it into 2π units. But naturally from a pure phenomenological consideration, the circle is the fundamental base unit of angle. That's why it's there. 

If I had to do it all over again, I would be very tempted to use circ as the base unit of angle. But then I caved to popular demand and used radian, 

> I think it is ultimately related to the question of whether angles are a unit (radians, degrees, revolutions) or a dimensionless ratio (which is likely what the metrologists would argue). Note that I was recently in a CIPM meeting where the issue of angular measurements was raised, and it was agreed that degrees/grads/radians are useful since they indicate a scaling factor on a dimensionless unit. In this context 'revolution' or 'circ' would likely be considered to be useful units.

No the RPM question is not ultimately related to the question if angles are dimension or dimensionless. I take some consolation for my intuition in these regards because I had predicted a decade before those "metrologists" finally agreed that a mol is nothing but Avogadro's number of particles (no more 12th of a g of Carbon 12 or whatever seemingly tautological it was, I was bad in 9th grade Chemistry because of that, so I have a personal vendetta, which I ended up winning, not because of my activism, but because [reason ultimately prevailed](https://iupac.org/new-definition-mole-arrived/)). In the same way, I bet that considering all units of angle as dimensionless units will ultimately fail. 

That 97% of metrologists agree today, doesn't impress me. It is ultimately arbitrary, and with the angle having taken in UCUM the vacant spot of the mole as a base unit, we stick to the lucky number 7.

**Comment by dr-shorthair** [comment]
> First of all, who is "Dr. Shorthair"? Is that Simon?

Yes - if you mouse-over or click on the picture that information is provided.

**Comment by colin-e-hscic** [comment]
There are some interesting papers around on these questions. A good one (to the extent I can understand it) from Dr. Paul Quincey at the UK National Physical Laboratory seems to come down on the same side as @gschadow, concluding that **_Radians_** and **_cycles_** are not dimensionless, but rather natural units of angle. 

It also has some interesting things to say about discrete (countable) quantities. I can probably dig out a copy if it's of interest.

**Comment by dr-shorthair** [comment]
Hmm. But if 'cycle' (`circ`?) is indeed a _unit_ of angle (natural or otherwise) then `circ/min` is a valid derived unit. 

I think @gschadow is arguing a slightly different point: that people who use RPM don't actually think of it as an angular velocity, but as a count-rate, so `min-1` is more meaningful.

**Comment by chgessner** [comment]
Also note related post https://github.com/orgs/ucum-org/discussions/259

**Comment by colin-e-hscic** [comment]
As a mechanical engineer by training (long ago) and long-time car and motorcycle enthusiast, I would have to disagree with the suggestion that "_people who use RPM don't actually think of it as an angular velocity, but as a count-rate_"

I've yet to meet a car or bike nut who is likely to think of his/her engine speed as "Hz", and I would say the same is true in most engineering contexts.

I do agree that rotation is strange in that it has two natural units that are just a small constant ratio apart. Radians works better for trigonometry (crudely speaking things in the range up to one full rotation), whereas cycles works better for multiple rotations and/or cyclic phenomena when the number of rotations is assumed to be infinite.

**Comment by colin-e-hscic** [comment]
In case these aren't in wide circulation (as a relative newbie in this area i'm wary of "teaching my granny to suck eggs") I have attached the comment from P. Quncey at NPL I referred to above, plus the Mohr and Philipps paper he refers to.

What I find interesting is at the end of the comment by Quincey he also pulls in the issue of dealing with that other class of dimensionless values- countable entities. Although I can't claim any insight into how these things could be implemented, it does seem there is a hint here of a general pattern of "named subclasses of a general class", whether it's named countable items, or named dimensionless ratios (e.g. "mass fraction") or named units of angle, that by default I imagine applications would want to preserve in calculations+conversions, but with the option to explicitly allow processing as a member of the superclass (if you know what you are doing).

A more advanced treatment might consider insisting that countable entities always have magnitudes that are integers, but that would start to pull UCUM out of its "only the units" space into being a more complete model of **_Quantity_**.
[Dimensionless Units in the SI- comment (PQ, NPL).pdf](https://github.com/ucum-org/ucum/files/10202244/Dimensionless.Units.in.the.SI-.comment.PQ.NPL.pdf)
[Mohr_2015_Metrologia_52_40.pdf](https://github.com/ucum-org/ucum/files/10202248/Mohr_2015_Metrologia_52_40.pdf)

**Comment by gschadow** [comment]
Dear @colin-e-hscic Thank you a bunch for providing the accessible links to those two papers.

About the RPM from the perspective of a motorcycle tinkerer, I don't think an actual engineer would be shocked by 60 RPM = 1 Hz, and measuring timing chain alignments in degrees and looking at oscilloscopes and thinking in Hz and kHz is second nature to an engineer. UCUM doesn't have to adopt jargon which less sophisticated people tend to speak. To avoid the scaling by 1/60, you can simply use the unit /min.

It is very interesting that both your cited authorities Mohr and Quincey agree that Hz is for rotational or cyclical phenomena only, which means they would say that 60 RPM = 1 Hz is the correct unit to be used for rotations (and I doubt they would be interested in hearing about the minute vs. second issue. 

Mohr would like us to use a unit "cycle" so that 1 RPM = 1 cycle/min. 

Both would say that Hz is not coherent and instead rad / s would be coherent, and hence 1 RPM should then be defined as and angular velocity ω = 2π/60 rad/s. Now our motorcycle mechanic's heads would blow up (although they might be calmed down by reminding them of timing chain adjustments.)

Since in UCUM I had jumped on the Mohr position with regards to angle as a base unit as clarified by Quincey, I had not appreciated the issue with radius length vs. length that this entails according to Quincey (and which I don't completely understand.) 

Both seem to agree that Hz should be demoted. 

I don't think from what has been presented so far there is any specific recommendation of what to do about RPM and that 1 RPM being just 1 /min  would not be good enough.

**Comment by gschadow** [comment]
PS: I would also be interested in Mohr's reference 6: https://iopscience.iop.org/article/10.1088/0026-1394/18/1/002/pdf

**Comment by colin-e-hscic** [comment]
My comment was aimed specifically at the question of whether "most users" (whatever that means) think of rpm as an angular velocity. In the broad sense of "how fast is this thing spinning?" then yes I think many of them do, at least in an engineering context,

I agree any engineer worth their salt will switch viewpoints to looking at rpm as describing a cyclic phenomenon when appropriate, e.g. when considering vibration, and the x60 factor to or from Hz is also not going to be an issue.

> About the RPM from the perspective of a motorcycle tinkerer, I don't think an actual engineer would be shocked by 60 RPM = 1 Hz, and measuring timing chain alignments in degrees and looking at oscilloscopes and thinking in Hz and kHz is second nature to an engineer. UCUM doesn't have to adopt jargon which less sophisticated people tend to speak. To avoid the scaling by 1/60, you can simply use the unit /min.


I have read the Quincey comment quite a few times and while most of it makes a lot of sense the "radius length" vs. "length" part does leave me baffled. I think at that point I just have to accept my lack of formal mathematical or metrological training leaves me unable to follow the argument or comment usefully, but I don't think that invalidates the rest of the case he makes.

**Comment by gschadow** [comment]
Me having the habit of singing over the hum of my motorcycle engine am perfectly happy about Hz. 

Perhaps if you and I form a study group we can figure out this mystery about the radius length yet. Meanwhile I could get talked into defining [RPM] = 1 /min.

**Comment by davidsummers** [comment]
Thanks for an interesting discussion/debate.

My "drive by" observation is that "RPM" means "1 revolution / min" so it is not "1 /min".   1 what?  1 Revolution.

1 Turn of the circle = 2 PI = 1 TAU

I disagree with the SI that angular measure is dimensionless. 

I think angular measure is a separate base dimension from anything else.

meter = linear length unit.

radian = angular measure unit, 1 TAU radians / circular unit in rad units.

steradian = volumetric angular measure in rad^2 units.

Here is how it can all fit together in better words and in more detail than I can think of.

https://iopscience.iop.org/article/10.1088/0026-1394/52/1/40

https://tauday.com/tau-manifesto

**Comment by gschadow** [comment]
Hi @davidsummers , I am excited that you agree with me (and UCUM) about the plain angle being a base dimension. 

Thank you also for adding another really intriguing reference to this τ-manifesto. It starts out hilarious and then goes into much interesting details and is overall thought provoking.

I think we were still struggling a bit to fully appreciate the Mohr paper's issue with angle vs. length.

I remember in the early days when UCUM came together I had pondered to use circle as the base unit. If we were to re-formulate UCUM bases based on natural constants, perhaps we would be doing that now.

But coming back to the original question: should "RPM" be considered an angular velocity (π/60 rad/s) or a frequency (1/60 Hz) is still the open question. I feel like Mohr et al. bashing Hz as an incoherent unit would perhaps give support to defining RPM as angular velocity. 

However, if you did that, someone else will come and say: now we want BPM (beats per minute) too!

Another oddity, which few people realize, is that angular velocity ω is actually a vector quantity, with the direction in the axis of the rotation. Do we really think that RPM folks understand that they have a vector quantity?

I have no final opinion. Would be curious what Christof @chgessner thinks about this.

**Comment by davidsummers** [comment]
@gschadow  Good point on the vector quantity.  My  best friend often points out that circular motion of sin/cos is really a 3D "helix" that describes a circle in 2D but a helix in 3D.

**Comment by chgessner** [comment]
The unit is not meant to discriminate the quantity being measured. It just gives us the scale. If the quantity is scalar, or a vector component (or even some tensor component) would be indicated in the context (e g LOINC code).
just my 2 c

**Comment by colin-e-hscic** [comment]
> 

Agreed the unit must align with (be commensurate with) the Quantity, but it often will not fully describe it. That can lead to problems of course if calculations are done that are "unit aware" but not "quantity aware". An example would be confusing the vector quantity _torque_ (N⨯m), with energy in Joules (N⋅m), because both can be described in units of _Nm_.

FWIW I agree with the proposal that RPM describes an angular velocity. For example RPM can be multiplied by an elapsed time and the result is a number of rotations. If the time period is small so the number of rotations comes out as less than 1, that can be directly converted into an angle of rotation in degrees/Radians etc.

Hz is a generic unit (perhaps a unit family or class) for "any cyclic phenomenon with a steady or slowly-changing repetition rate". You can describe anything from the frequency of UV light to the human birth rate on earth or the orbit of the earth around the sun in Hz (even if the numeric values are a bit awkward). The dimension of the Quantity measured on a Hz scale depends on what it is that is repeating, the numerator in (something)/s

**Comment by colin-e-hscic** [comment]
> Me having the habit of singing over the hum of my motorcycle engine am perfectly happy about Hz.
> 
> Perhaps if you and I form a study group we can figure out this mystery about the radius length yet. Meanwhile I could get talked into defining [RPM] = 1 /min.

I have spoken to Dr. Quincey in the past, so I might even be able to persuade him to explain it to us personally.

There's a practical issue here in that I am only weeks away from retirement myself, and when I leave I'll lose my NHS Digital SSO login ID on GitHub. I may disappear entirely, or if I decide to keep up the conversation I'll have to reappear under one of my other Github IDs, when presumably I will need to update my profile to be allowed back into the fold ;-)

If nothing else as a long-time motorcyclist I'm curious to know what bike you run ;-)
